kornia/image/base.py

```diff
@@ -11,13 +10,58 @@ class ImageSize:
 
     Args:
         height: image height.
-        width: image width
+        width: image width.
+
     Example:
         >>> size = ImageSize(3, 4)
         >>> size.height
         3
         >>> size.width
         4
     """
-    height: int | Tensor
-    width: int | Tensor
+    height: int
+    width: int
+
+
+class PixelFormat(Enum):
+    r"""Enum that represents the pixel format of an image."""
+    GRAY = 0
+    RGB = 1
+    BGR = 2
+
+
+class ChannelsOrder(Enum):
+    r"""Enum that represents the channels order of an image."""
+    CHANNELS_FIRST = 0
+    CHANNELS_LAST = 1
+
+
+@dataclass
+class ImageLayout:
```

Comment:
If we don't expect to change ImageLayout in-place, perhaps we can consider either `@dataclass(frozen=True)` or namedtuple instead?

Reply:
maybe it can change, e.g if we want to support from/to LAST/FISR channel, or resize, etc

Reply:
I don't have a strong preference on this, so the current way is also fine. I'm just thinking, when an Image is transformed (e.g. change channel format, resize...), we will likely create a new Tensor object. It may not make much sense to re-use the old ImageLayout object (and the Image object) instead of just creating a new one.